README.md

```diff
@@ -1,28 +1,71 @@
+This repo helps setting up the CI environment used for executing Snappy integration tests [1]. In contains a Dockerfile which defines the configuration of a Jenkins container and several shell scripts for bringing it to live both in a cloud provider (only OpenStack supported at the moment) and locally.
 
-You can use this container locally or in a cloud instance, there are provision script for either case (see below). In order to use the container properly you need to have set up an openstack environment, with at least the $OS_USERNAME and $OS_REGION variables set. There's also some required packages:
+The testing jenkins jobs use use the binary from this project [2] for launching the tests, it takes care of determining the most recent Snappy cloud image for a given channel and release (see below for the required name patterns), instantiating a cloud instance of that image and getting its ip, executing the integration suite in the instance and shutting it down when finished or in case of errors.
 
-    $ sudo apt-get install cloud-utils python-novaclient
+There are also jobs configured for accessing the Snappy Product Integration environment. This gives a lot more flexibility when executing the tests not being only limited to cloud instances, however you must provide credentials for this service.
 
 ## Provision
 
-There are two provision scripts. Both assume that you have openstack credentials loaded, for instance having sourced a novarc file.
+There are provision scripts for creating the CI environment locally and in the cloud. In both cases you need to have OpenStack credentials loaded, for example by sourcing an novarc file:
+
+    $ source path/to/openstack/credentials/novarc
+
+Before Executing the provision script you should have at least the ```$OS_USERNAME``` and ```$OS_REGION_NAME``` environment variables set.
+
+### Local Provision
+
+You can setup the enviroment locally by executing this command:
+
+    $ ./bin/local-provision <cloud_credentials_path>
+
+The ```<cloud_credentials_path>``` required by the command indicates a path where OpenStack credentials to be used by the Jenkins instance to spin up Snappy instances can be found. It requires a novarc file and an user key file in that directory, the scripts copies them to the Jenkins container and the jobs use them to access to the cloud provider API.
+
+Once the scripts finishes you can access the jenkins instance from the browser at ```http://localhost:8080```
+
+### Cloud provision
+
+There are additional requirements, beside having loaded OpenStack credentials, for the cloud provision, in this case the setup is done in a cloud instance and you should be able to spin it up, the needed packages can be installed in Ubuntu with:
```

Comment:
many sentences here separated by commas.
There are additional requirements for the cloud provision; besides having loaded Openstack credentials. In this case the setup is done in a cloud instance and you should be able to spin it up. The needed packages can be installed in Ubuntu with:

sounds better?